src/sql/src/statements/transform/expand_interval.rs

```diff
@@ -137,26 +138,62 @@ fn update_existing_interval_with_value(interval: &Interval, value: Box<Expr>) ->
     })
 }
 
-/// Expands an interval abbreviation to its full name.
+/// Normalizes an interval expression string into the sql-compatible format.
+/// This function handles 2 types of input:
+/// 1. Abbreviated interval strings (e.g., "1y2mo3d")
 /// Returns an interval's full name (e.g., "years", "hours", "minutes") according to the `INTERVAL_ABBREVIATION_MAPPING`
 /// If the `interval_str` contains whitespaces, the interval name is considered to be in a full form.
-/// Hybrid format "1y 2 days 3h" is not supported.
-fn expand_interval_name(interval_str: &str) -> Option<String> {
-    return if !interval_str.contains(|c: char| c.is_whitespace()) {
-        Some(
-            INTERVAL_ABBREVIATION_PATTERN
-                .find_iter(interval_str)
-                .map(
-                    |mat| match INTERVAL_ABBREVIATION_MAPPING.get(mat.as_str()) {
-                        Some(&expanded_name) => expanded_name,
-                        None => mat.as_str(),
-                    },
-                )
-                .join(" "),
-        )
+/// 2. ISO 8601 format strings (e.g., "P1Y2M3D")
+/// Returns a number of milliseconds corresponding to ISO 8601 (e.g., "36525000 milliseconds")
+/// Note: Hybrid format "1y 2 days 3h" is not supported.
+fn normalize_interval_name(interval_str: &str) -> Option<String> {
+    if interval_str.contains(char::is_whitespace) {
+        return None;
+    }
+
+    if is_iso8601_format(interval_str) {
+        return parse_iso8601_interval(interval_str);
+    }
+
+    expand_interval_abbreviation(interval_str)
+}
+
+fn parse_iso8601_interval(signed_iso: &str) -> Option<String> {
+    let (is_negative, unsigned_iso) = if let Some(stripped) = signed_iso.strip_prefix('-') {
+        (true, stripped)
     } else {
-        None
+        (false, signed_iso)
     };
+
+    match iso8601::duration(&unsigned_iso.to_uppercase()) {
+        Ok(duration) => {
+            let millis = StdDuration::from(duration).as_millis();
+            let sign = if is_negative { "-" } else { "" };
+            Some(format!("{}{} milliseconds", sign, millis))
+        }
+        Err(_) => None,
+    }
+}
+
+fn expand_interval_abbreviation(interval_str: &str) -> Option<String> {
+    Some(
+        INTERVAL_ABBREVIATION_PATTERN
+            .find_iter(interval_str)
+            .map(|mat| {
+                let mat_str = mat.as_str();
+                *INTERVAL_ABBREVIATION_MAPPING
+                    .get(mat_str)
+                    .unwrap_or(&mat_str)
+            })
+            .join(" "),
+    )
+}
+
+fn is_iso8601_format(interval_str: &str) -> bool {
+    interval_str.starts_with('P')
```

Comment:
What about using a regex? Looks more clear.

Reply:
Good point indeed! I was adding conditions one by one (lowercase case, then negative and forgot to review with the fresh eye)